src/Aspire.Hosting.Azure.ApplicationInsights/AzureApplicationInsightsExtensions.cs

```diff
@@ -76,15 +77,26 @@ public static IResourceBuilder<AzureApplicationInsightsResource> AddAzureApplica
 
             if (logAnalyticsWorkspace != null)
             {
+                // If someone provides a workspace via the extension method we should use it.
                 appInsights.AssignProperty(p => p.WorkspaceResourceId, logAnalyticsWorkspace.Resource.WorkspaceId, AzureBicepResource.KnownParameters.LogAnalyticsWorkspaceId);
             }
-            else
+            else if (builder.ExecutionContext.IsRunMode)
             {
+                // ... otherwise if we are in run mode, the provisioner expects us to create one ourselves.
+                var autoInjectedLogAnalyticsWorkspace = new OperationalInsightsWorkspace(construct, name: $"law-{construct.Resource.Name}");
+                appInsights.Properties.Tags["aspire-resource-name"] = $"law-{construct.Resource.Name}";
```

Comment:
Is this correct? Is it supposed to be:

```suggestion
                autoInjectedLogAnalyticsWorkspace.Properties.Tags["aspire-resource-name"] = $"law-{construct.Resource.Name}";
```

Also, can we remove the duplicate `$"law-{construct.Resource.Name}"` and just use `autoInjectedLogAnalyticsWorkspace.Name` to reuse the value we set on line 86?

Reply:
That name field is the bicep construct name. And the other one would be an expression. 3 different values. But I can tweak this a bit so I only generate the string once.